xyz-hub-service/src/main/java/com/here/xyz/hub/task/ModifyFeatureOp.java

```diff
@@ -37,19 +37,47 @@
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
+import java.util.function.Function;
 import java.util.stream.Collectors;
+import java.util.stream.Stream;
 
 public class ModifyFeatureOp extends ModifyOp<Feature, FeatureEntry> {
 
-  public ModifyFeatureOp(List<Map<String, Object>> inputStates, IfNotExists ifNotExists, IfExists ifExists, boolean isTransactional, ConflictResolution cr) {
-    super((inputStates == null) ? Collections.emptyList() : inputStates.stream().map(m-> new FeatureEntry(m,cr)).collect(Collectors.toList()),
-        ifNotExists, ifExists, isTransactional);
+  private final static String ON_FEATURE_NOT_EXISTS = "onFeatureNotExists";
+  private final static String ON_FEATURE_EXISTS = "onFeatureExists";
+  private final static String ON_MERGE_CONFLICT = "onMergeConflict";
+
+  /**
+   *
+   * @param modificationInput A list of FeatureCollections of which each may have different settings
+   *  for existence-handling and/or conflict-handling.
+   *  If these settings are not specified at the FeatureCollection the according other parameters (ifNotExists, ifExists, conflictResolution)
+   *  of this constructor will be applied for that purpose.
+   * @param ifNotExists
+   * @param ifExists
+   * @param isTransactional
+   * @param conflictResolution
+   */
+  public ModifyFeatureOp(List<Map<String, Object>> modificationInput, IfNotExists ifNotExists, IfExists ifExists, boolean isTransactional,
+      ConflictResolution conflictResolution) {
+    super((modificationInput == null) ? Collections.emptyList() : modificationInput.stream().flatMap(fc -> {
+      IfNotExists ne = fc.get(ON_FEATURE_NOT_EXISTS) instanceof String ? IfNotExists.of((String) fc.get(ON_FEATURE_NOT_EXISTS)) : ifNotExists;
+      IfExists e = fc.get(ON_FEATURE_EXISTS) instanceof String ? IfExists.of((String) fc.get(ON_FEATURE_EXISTS)) : ifExists;
+      ConflictResolution cr = fc.get(ON_MERGE_CONFLICT) instanceof String ?
+          ConflictResolution.of((String) fc.get(ON_MERGE_CONFLICT)) : conflictResolution;
+
+      List<Map<String, Object>> features = (List<Map<String, Object>>) fc.get("features");
+      if (features == null)
+        return Stream.empty();
+
+      return features.stream().map(feature -> new FeatureEntry(feature, ne, e, cr));
+    }).collect(Collectors.toList()), ifNotExists, ifExists, isTransactional);
```

Comment:
This can be removed since the logic migrated to the Entry.
ifNotExists and ifExists are not necessary anymore in the ModifyOp.

Reply:
I removed the properties from ModifyOp in my latest commit.
Removing these global flags is not only consistent but also prevented a security-hole so thanks a lot for your hint.